lib/storage/storage.go

```diff
@@ -1983,7 +1963,12 @@ func (s *Storage) storeTSIDToCaches(metricNameRaw []byte, genTSID *generationTSI
 	s.dateMetricIDCache.Set(genTSID.generation, date, genTSID.TSID.MetricID)
 }
 
-func (s *Storage) registerSeriesCardinality(metricID uint64, metricNameRaw []byte) bool {
+func (s *Storage) registerSeriesCardinality(metricNameRaw []byte) bool {
+	if s.hourlySeriesLimiter == nil && s.dailySeriesLimiter == nil {
+		return true
+	}
+
+	metricID := xxhash.Sum64(metricNameRaw)
```

Comment:
is there a chance to get a collision on this?

Reply:
Yes, it's possible. But vmagent uses the same hash function for cardinality limiting.